src/researches/getPassiveVoice.js

```diff
@@ -1,110 +1,122 @@
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "fr" ) );
-			break;
-		case "es":
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "es" ) );
-			break;
-		case "en":
-		default:
-			sentencePart.setPassive( determinePassivesDefault( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), "en" ) );
-			break;
+let getPassivesPeriphrastic = function( sentenceObjects, language ) {
+	let passiveSentencesByType = [];
+	forEach( sentenceObjects, function( sentence ) {
+		let strippedSentence = stripHTMLTags( sentence.getSentenceText() ).toLocaleLowerCase();
+		let sentenceParts = [];
+
+		if ( language === "de" ) {
+			sentenceParts = getSentencePartsGerman( strippedSentence );
+		} else {
+			sentenceParts = getSentencePartsDefault( strippedSentence, language );
+		}
+
+		let passive = false;
+		forEach( sentenceParts, function( sentencePart ) {
+			sentencePart.setPassive( determinePassiveSentencePart( sentencePart.getSentencePartText(), sentencePart.getAuxiliaries(), language ) );
+			passive = passive || sentencePart.isPassive();
+		} );
+		if ( passive === true ) {
+			passiveSentencesByType.push( sentence.getSentenceText() );
+		}
+	} );
+	return {
+		passiveSentencesByType,
+	};
+};
+
+/**
+ * Determines which passive voice marking should be executed for a text based on the language.
+ *
+ * @param {Array} sentenceObjects Sentences extracted from the text.
+ * @param {string} language Language of the text.
+ * @returns {object} The number of passives found in the text and the passive sentences.
+ */
+let getPassives = function( sentenceObjects, language ) {
+	let passiveSentences = [];
+
+	if ( morphologicalLanguages.includes( language ) ) {
+		passiveSentences = getPassivesMorphological( sentenceObjects, language );
+	} else if ( periphrasticLanguages.includes( language ) ) {
+		passiveSentences = getPassivesPeriphrastic( sentenceObjects, language );
+	} else if ( morphologicalAndPeriphrasticLanguages.includes( language ) ) {
+		passiveSentences = getPassivesMorphological( sentenceObjects, language );
+		passiveSentences.push( getPassivesPeriphrastic( sentenceObjects, language ) );
 	}
+
+	return {
+		passiveSentences,
+	};
 };
 
+
 /**
  * Determines the number of passive sentences in the text.
  *
  * @param {Paper} paper The paper object to get the text from.
  * @returns {object} The number of passives found in the text and the passive sentences.
  */
 module.exports = function( paper ) {
-	var text = paper.getText();
-	var locale = paper.getLocale();
-	var language = getLanguage( locale );
-	var sentences = getSentences( text );
+	let text = paper.getText();
+	let locale = paper.getLocale();
+	let language = getLanguage( locale );
+	let sentences = getSentences( text );
 
-	var sentenceObjects = [];
+	let sentenceObjects = [];
 
 	forEach( sentences, function( sentence ) {
```

Comment:
To facilitate changing the name of `sentenceObjects` to `sentences` you can write this forEach like this:

```js
const sentences = getSentences( text )
    .map( sentence => new Sentence( sentence ) );

// To make clearer what is happening there you can fully write out the arrow function:
const sentences = getSentences( text )
    .map( ( sentence ) => { new Sentence( sentence ) } );

// Of if that is not clear, like this:
const sentences = getSentences( text )
    .map( function( sentence ) { return new Sentence( sentence ); } );
```